plugins/parquet/parquetembed.cpp

```diff
@@ -956,12 +956,42 @@ __int64 ParquetRowBuilder::getCurrIntValue(const RtlFieldInfo *field)
         {
             __int64 myint64 = 0;
             auto scalar = getCurrView(field);
-            handleDeserializeOutcome(tokenDeserializer.deserialize(scalar.data(), myint64), "signed", scalar.data());
+            handleDeserializeOutcome(tokenDeserializer.deserialize(StringBuffer(scalar.size(), scalar.data()).str(), myint64), "signed", scalar.data());
             return myint64;
         }
     }
 }
 
+double ParquetRowBuilder::getCurrRealValue(const RtlFieldInfo *field)
+{
+    switch ((*array_visitor)->type)
+    {
+        case BoolType:
+            return (*array_visitor)->bool_arr->Value(currArrayIndex());
+        case IntType:
+            return getSigned(array_visitor, currArrayIndex());
+        case UIntType:
+            return getUnsigned(array_visitor, currArrayIndex());
+        case RealType:
+            return getReal(array_visitor, currArrayIndex());
+        case DateType:
+            return (*array_visitor)->size == 32 ? (*array_visitor)->date32_arr->Value(currArrayIndex()) : (*array_visitor)->date64_arr->Value(currArrayIndex());
+        case TimestampType:
+            return (*array_visitor)->timestamp_arr->Value(currArrayIndex());
+        case TimeType:
+            return (*array_visitor)->size == 32 ? (*array_visitor)->time32_arr->Value(currArrayIndex()) : (*array_visitor)->time64_arr->Value(currArrayIndex());
+        case DurationType:
+            return (*array_visitor)->duration_arr->Value(currArrayIndex());
+        default:
+        {
+            double mydouble = 0.0;
+            auto scalar = getCurrView(field);
+            handleDeserializeOutcome(tokenDeserializer.deserialize(StringBuffer(scalar.size(), scalar.data()).str(), mydouble), "real", scalar.data());
```

Comment:
minor: as per prev. comment - consider declaring temp string container on previous line and using std::string.

Reply:
Changed.